### PR TITLE
CLN: trim whitespace in ipynb/py files in parallel

### PR DESCRIPTION
**Pull request by jbrockmendel**
This gets the bits that can be gotten directly in the ipynb files.  For some of it I now need to look at nbgenerate.py and see where extraneous whitespace is getting added.

Also removed some unnecessary semicolons.

Longer-term I'm pretty sure the way to keep these in sync is to only generate the py files at test+release time.

### CODE REVIEW

File: examples/notebooks/robust_models_1.ipynb [hunk 1]
```diff
@@ -263,7 +263,7 @@
     "c = 4.685\n",
     "support = np.linspace(-3*c, 3*c, 1000)\n",
     "tukey = norms.TukeyBiweight(c=c)\n",
-    "plot_weights(support, tukey.weights, ['-3*c', '0', '3*c'], [-3*c, 0, 3*c]);"
```
Comment (josef-pkt) [inline examples/notebooks/robust_models_1.ipynb]: those semicolons are INTENTIONAL.
they are to prevent the plots from being rendered twice

**Comment by josef-pkt** [comment]
Did you edit the notebooks files directly or through jupyter?
that is: Is the whitespace created by users or by the notebook format and py conversion?

**Comment by jbrockmendel** [comment]
> Did you edit the notebooks files directly or through jupyter?

Manually edited both side-by-side to ensure all changes are parallel.  I recognize this is not trivial to verify manually.

Currently trying to figure out how to generate .py file from .ipynb file.  I had assumed tools/nbgenerate.py was for that, but that seems to only want to produce .rst or .html.  `jupyter nbconvert robust_models_0.ipynb --to=script` produces a _similar_ .py file but has a bunch of ipython REPL added in.  What's the canonical way sm produces these?

**Comment by josef-pkt** [comment]
I don't know, I looked at it quite some time ago (years?), and didn't see any simple automatic way to export.
I always export manually, i.e. directly from a running notebook after cleaning cells (removing all output).

File: examples/notebooks/robust_models_1.ipynb (hunk 1 above)
Comment (jbrockmendel) [reply examples/notebooks/robust_models_1.ipynb]: How does that work?  How would matplotlib even know there's a semicolon?

**Comment by jbrockmendel** [comment]
>I don't know, I looked at it quite some time ago (years?), and didn't see any simple automatic way to export.  I always export manually, i.e. directly from a running notebook after cleaning cells (removing all output).

OK.  Definitely not a high-priority, but automating this should be somewhere on the todo list.

File: examples/notebooks/robust_models_1.ipynb (hunk 1 above)
Comment (josef-pkt) [reply examples/notebooks/robust_models_1.ipynb]: It's not matplotlib, it's jupyter
https://github.com/statsmodels/statsmodels/issues/1265#issuecomment-383336489

**Comment by josef-pkt** [comment]
The best would be a post-commit hook to export notebook to py, but last time I checked, those don't work on Windows.

File: examples/notebooks/robust_models_1.ipynb (hunk 1 above)
Comment (jbrockmendel) [reply examples/notebooks/robust_models_1.ipynb]: Well that's a kick in the teeth.

I think the thing to do here is to close this PR and put together a robust long-term git-hook-resembling solution that keeps the semicolon in the .ipynb version and removes it in the .py version.  (and to encourage you to merge #5036)

**Pull request closed without merging**